src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -214,7 +216,37 @@ SpecialType.System_Collections_Generic_IReadOnlyCollection_T or
                     TypeArgumentsWithAnnotationsNoUseSiteDiagnostics: [var elementType]
                 })
             {
-                arrayOrList = CreateEmptyArray(node.Syntax, ArrayTypeSymbol.CreateSZArray(_compilation.Assembly, elementType));
+                var elements = node.Elements;
+                int length = elements.Length;
+                if (length == 0)
+                {
+                    arrayOrList = CreateEmptyArray(syntax, ArrayTypeSymbol.CreateSZArray(_compilation.Assembly, elementType));
+                }
+                else
+                {
+                    Debug.Assert(!elements.Any(e => e is BoundCollectionExpressionSpreadElement)); // PROTOTYPE: Not handled.
+
+                    var synthesizedType = _factory.ModuleBuilderOpt.EnsureReadOnlyListTypeExists(syntax, _diagnostics.DiagnosticBag).Construct(ImmutableArray.Create(elementType));
+
+                    // var array = new ElementType[] { e1, ..., eN };
+                    var initialization = new BoundArrayInitialization(
+                        syntax,
+                        isInferred: false,
+                        elements.SelectAsArray(e => VisitExpression(e)));
+                    var arrayType = ArrayTypeSymbol.CreateSZArray(_compilation.Assembly, elementType);
```

Comment:
nit: consider inlining this to make the code seem more like it's pure declarative construction.